ci/Dockerfile

```diff
@@ -0,0 +1,76 @@
+# Taichi Dockerfile for development
+ARG UBUNTU
+FROM nvidia/cuda:${UBUNTU}
+ARG PYTHON
+ARG TEST_OPTION
+ARG PYPI_PWD
+ENV PYPI_PWD=$PYPI_PWD
+ENV DEBIAN_FRONTEND=noninteractive
+LABEL maintainer="https://github.com/taichi-dev"
+
+# This installs Python 3.6.9 by default. Once the
+# docker image is upgraded to Ubuntu 20.04 this will
+# install Python 3.8 by default
+RUN apt-get update && \
+    apt-get install -y software-properties-common \
+                       $PYTHON \
+                       python3-pip \
+                       ${PYTHON}-dev\
+                       libtinfo-dev \
+                       clang-10 \
+                       wget \
+                       git \
+                       libx11-dev \
+                       libxrandr-dev \
+                       libxinerama-dev \
+                       libxcursor-dev \
+                       libxi-dev \
+                       libglu1-mesa-dev \
+                       freeglut3-dev \
+                       mesa-common-dev \
+                       build-essential \
+                       libssl-dev \
+                       libidn11-dev \
+                       libz-dev \
+                       unzip
+
+# Install the latest version of CMAKE v3.20.2 from source
+WORKDIR /
+RUN wget https://github.com/Kitware/CMake/releases/download/v3.20.5/cmake-3.20.5-linux-x86_64.tar.gz
+RUN tar xf cmake-3.20.5-linux-x86_64.tar.gz
+ENV PATH="/cmake-3.20.5-linux-x86_64/bin:$PATH"
+
+# Intall LLVM 10
+WORKDIR /
+RUN wget https://github.com/taichi-dev/taichi_assets/releases/download/llvm10_linux/taichi-llvm-10.0.0-linux.zip
+RUN unzip taichi-llvm-10.0.0-linux.zip
+ENV PATH="/taichi-llvm-10.0.0-linux/bin:$PATH"
+
+# Install Taichi from source
+ENV CC="clang-10"
+ENV CXX="clang++-10"
+WORKDIR /taichi-dev
+RUN git clone https://github.com/taichi-dev/taichi --depth=1 --branch=master
+# Install Taichi's Python dependencies
+RUN $PYTHON -m pip install --user -r taichi/requirements.txt
+# Build Taichi from source
+RUN cd taichi && \
+    git submodule update --init --recursive --depth=1 && \
+    mkdir build
+WORKDIR /taichi-dev/taichi/build
+RUN cmake .. -DPYTHON_EXECUTABLE=$PYTHON -DCUDA_VERSION='10.0' -DTI_WITH_CUDA:BOOL=True -DTI_WITH_CC:BOOL=ON -DCMAKE_CXX_COMPILER=$CXX
```

Comment:
`-DCUDA_VERSION='10.0'`is related to the docker tag, maybe parameterize it too?

Reply:
I believe this is just the hacked version we pass in order to avoid build errors.

Reply:
Right, we kind of have to pass in 10.0 for `-DCUDA_VERSION` (even if the actual CUDA version is 11.0) :-/

Reply:
Ah gotcha, thanks for the explanation!